project/src/main/puzzle/food-item.gd

```diff
@@ -79,9 +79,12 @@ func _physics_process(delta: float) -> void:
 			_source_pos += velocity * delta
 			_target_pos = _get_target_pos.call_funcv(_target_pos_arg_array)
 			
-			# The y coordinate changes at a constant rate while the x coordinate follows a parabolic path
-			position.y = lerp(_source_pos.y, _target_pos.y, _flight_percent)
-			position.x = lerp(_source_pos.x, _target_pos.x, 2.1 * pow(_flight_percent, 1.6) - 1.1 * _flight_percent)
+			if _target_pos == Vector2.INF:
+				queue_free()
```

Comment:
This is worth a comment, it's very surprising.

Reply:
Added a comment, `get_target_pos returns Vector2.INF if the food should be removed`